github/util.py

```diff
@@ -221,3 +224,95 @@ def retrieve_email(username: str):
         )
     )
 
+def _create_team(
+    github: GitHub,
+    organization_name: str,
+    team_name: str
+):
+    # passed GitHub object must have org. admin authorization to create a team
+    organization = github.organization(organization_name)
+    team = _retrieve_team_by_name(organization, team_name)
+    if team:
+        util.verbose("Team {name} already exists".format(name=team_name))
+        return
+
+    try:
+        organization.create_team(name=team_name)
+        util.info("Team {name} created".format(name=team_name))
+    except ForbiddenError as err:
+        util.fail("{err} Cannot create team {name} in org {org} due to missing privileges".format(
+            err=err,
+            name=team_name,
+            org=organization_name
+        ))
+
+def _add_user_to_team(
+    github: GitHub,
+    organization_name: str,
+    team_name: str,
+    user_name: str
+):
+    # passed GitHub object must have org. admin authorization to add a user to a team
+    organization = github.organization(organization_name)
+    team = _retrieve_team_by_name(organization, team_name)
+    if not team:
+        util.fail("Team {name} does not exist".format(name=team_name))
+
+    if team.is_member(user_name):
+        util.verbose("{username} is already assigned to team {teamname}".format(
+            username=user_name,
+            teamname=team_name
+        ))
+        return
+
+    if team.add_member(username=user_name):
+        util.info("Added {username} to team {teamname}".format(
+            username=user_name,
+            teamname=team_name
+        ))
+    else:
+        util.fail("Could not add {username} to team {teamname}. Please check for missing privileges".format(
+            username=user_name,
+            teamname=team_name
+        ))
+
+
+class RepoPermission(Enum):
+    pull = "pull"
+    push = "push"
+    admin = "admin"
+
+
+def _add_all_repos_to_team(
+    github: GitHub,
+    organization_name: str,
+    team_name: str,
+    permission: RepoPermission=RepoPermission.admin
+):
+    # passed GitHub object must have org. admin authorization to assign team to repo with admin rights
+    organization = github.organization(organization_name)
+    team = _retrieve_team_by_name(organization, team_name)
+    if not team:
+        util.fail("Team {name} does not exist".format(name=team_name))
+
+    for repo in organization.repositories():
+        if team.has_repository(repo.full_name):
+            util.verbose("Team {teamnname} already assigned to repo {reponame}".format(
+                teamnname=team_name,
+                reponame=repo.full_name
+            ))
+            continue
+
+        team.add_repository(repository=repo.full_name, permission=permission.value)
+        util.info("Added team {teamname} to repository {reponame}".format(
+            teamname=team_name,
+            reponame=repo.full_name
+        ))
+
+def _retrieve_team_by_name(
```

Comment:
_could_ be more expressive, if it '_or_none' or 'if_present' or sth. similar were appended to the name. also, it would be nice to state the contract in a docstring (the fact that None is returned if the team does not exist)